src/main/java/me/clip/placeholderapi/PlaceholderAPI.java

```diff
@@ -43,20 +43,50 @@
 
 public final class PlaceholderAPI {
 
-  private static final Replacer REPLACER_PERCENT = new CharsReplacer(Closure.PERCENT);
-  private static final Replacer REPLACER_BRACKET = new CharsReplacer(Closure.BRACKET);
-
   private static final Pattern PLACEHOLDER_PATTERN = Pattern.compile("[%]([^%]+)[%]");
   private static final Pattern BRACKET_PLACEHOLDER_PATTERN = Pattern.compile("[{]([^{}]+)[}]");
   private static final Pattern RELATIONAL_PLACEHOLDER_PATTERN = Pattern
       .compile("[%](rel_)([^%]+)[%]");
 
 
-  private PlaceholderAPI() {
-  }
+  private PlaceholderAPI() {}
 
   // === Current API ===
 
+
+  /**
+   * Translates all placeholders into their corresponding values.
+   * <br>The pattern of a valid placeholder depends on the {@link Closure} type provided.
+   *
+   * @param player Player to parse the placeholders against
+   * @param text Text to set the placeholder values in
+   * @param closure Closure defining the placeholder type and parsing logic
+   * @return String containing all translated placeholders
+   */
+  @NotNull
+  public static String getWithPlaceholders(final OfflinePlayer player,
+                                           @NotNull final String text,
+                                           Closure closure) {
+    return closure.getReplacer().apply(text, player,
+            PlaceholderAPIPlugin.getInstance().getLocalExpansionManager()::getExpansion);
+  }
+
+  /**
+   * Translates all placeholders into their corresponding values for a list of strings.
+   * <br>The pattern of a valid placeholder depends on the {@link Closure} type provided.
+   *
+   * @param player Player to parse the placeholders against
+   * @param text List of strings to set the placeholder values in
+   * @param closure Closure defining the placeholder type and parsing logic
+   * @return List of strings containing all translated placeholders
+   */
+  @NotNull
+  public static List<String> getWithPlaceholders(final OfflinePlayer player,
+                                                 @NotNull final List<String> text,
+                                                 Closure closure) {
+    return text.stream().map(line -> getWithPlaceholders(player, line, closure)).collect(Collectors.toList());
+  }
+
```

Comment:
What even is the benefit of these, if there is any?

All this does is add the requirement of providing the Closure yourself. And given there only is percent and bracket options, is this nothing different than the already existing methods outside of adding extra work.